### PR TITLE
Fix NsJail Tests

### PR DESCRIPTION
**Pull request by MarkKoz**
#88 left tests in a broken state. This wasn't noticed since CI had a bug that prevented the step from failing despite tests failing. This PR fixes both of these issues.

### CODE REVIEW

**Review by Akarys42 — APPROVED** [review]
LGTM! (except for the failing test)

**Comment by MarkKoz** [comment]
Yeah this wasn't ready for review. I became too busy to continue and fix those issues.

File: .github/workflows/lint-test-build-push.yaml [hunk 1]
```diff
@@ -150,8 +143,8 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
-          pip install coveralls~=2.1
-          coveralls
+          sudo pip install coveralls~=2.1
+          sudo coveralls
```
Comment (Akarys42) [inline .github/workflows/lint-test-build-push.yaml:147]: Is there any reason why we don't make a user install rather than a system install?

Comment (MarkKoz) [reply .github/workflows/lint-test-build-push.yaml:147]: IIRC it's cause the user site is not on PATH. It's more convenient to use `sudo` in this case.

Comment (MarkKoz) [reply .github/workflows/lint-test-build-push.yaml:147]: Though maybe it's bad practice, even in CI, and I should have kept using the Python version task...

Comment (MarkKoz) [reply .github/workflows/lint-test-build-push.yaml]: Don't yell at me, but I forced push to get rid of it. Turns out I had already dropped the commit locally last month, forgot about it, and commited changes over it.

**Comment by MarkKoz** [comment]
@jb3 One of the tests is failing. This is a similar issue to what #94 was trying to fix, but the test still fails despite merging that into this branch.

```
======================================================================
ERROR: test_stdout_flood_results_in_graceful_sigterm (tests.test_nsjail.NsJailTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/snekbox/snekbox/tests/test_nsjail.py", line 188, in test_stdout_flood_results_in_graceful_sigterm
    result = self.nsjail.python3(stdout_flood)
  File "/home/runner/work/snekbox/snekbox/snekbox/nsjail.py", line 236, in python3
    child.rmdir()
  File "/usr/local/lib/python3.9/pathlib.py", line 1352, in rmdir
    self._accessor.rmdir(self)
OSError: [Errno 16] Device or resource busy: '/sys/fs/cgroup/memory/snekbox-976324a6-c789-49eb-b88e-4702e4ac0626/NSJAIL.901'
```

Do you have any ideas?

File: .github/workflows/lint-test-build-push.yaml (hunk 1 above)
Comment (Akarys42) [reply .github/workflows/lint-test-build-push.yaml]: Cool!

**Comment by jb3** [comment]
So I guess the child cgroups are still busy. I'll check it out.

**Comment by MarkKoz** [comment]
Turns out that killing the parent process does not kill its children. Therefore, we have nsjail's children (which are Python processes in this case) sticking around preventing child cgroups from being removed. This will be addressed in a separate PR.

**Pull request merged**